recipes/approvaltests.cpp/7.0.0/conanfile.py

```diff
@@ -48,11 +49,16 @@ def package(self):
 
     def package_info(self):
         if self.options.test_framework == "catch2":
-            self.cpp_info.defines = ["APPROVALS_CATCH"]
+            """ Do not define APPROVALS_CATCH here.
```

Comment:
Do you have a log representing your comment? If I understand, you should define APPROVALS_CATCH only once, is that?

Reply:
If I have multiple test files that `#include <ApprovalTests.hpp>` I end up with errors like this:

```
----- snip -----
CMakeFiles/test_main.dir/hello_at.cpp.o: In function `Catch::XmlReporter::testCaseEnded(Catch::TestCaseStats const&)':
/home/dheater/.conan/data/Catch2/2.11.0/catchorg/stable/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/catch2/catch.hpp:17051: multiple definition of `Catch::XmlReporter::testCaseEnded(Catch::TestCaseStats const&)'
CMakeFiles/test_main.dir/main_at.cpp.o:/home/dheater/.conan/data/Catch2/2.11.0/catchorg/stable/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/catch2/catch.hpp:17051: first defined here
CMakeFiles/test_main.dir/hello_at.cpp.o: In function `Catch::XmlReporter::testGroupEnded(Catch::TestGroupStats const&)':
/home/dheater/.conan/data/Catch2/2.11.0/catchorg/stable/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/catch2/catch.hpp:17067: multiple definition of `Catch::XmlReporter::testGroupEnded(Catch::TestGroupStats const&)'
CMakeFiles/test_main.dir/main_at.cpp.o:/home/dheater/.conan/data/Catch2/2.11.0/catchorg/stable/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/catch2/catch.hpp:17067: first defined here
CMakeFiles/test_main.dir/hello_at.cpp.o: In function `Catch::XmlReporter::testRunEnded(Catch::TestRunStats const&)':
/home/dheater/.conan/data/Catch2/2.11.0/catchorg/stable/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/catch2/catch.hpp:17077: multiple definition of `Catch::XmlReporter::testRunEnded(Catch::TestRunStats const&)'
CMakeFiles/test_main.dir/main_at.cpp.o:/home/dheater/.conan/data/Catch2/2.11.0/catchorg/stable/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/catch2/catch.hpp:17077: first defined here
CMakeFiles/test_main.dir/hello_at.cpp.o:(.bss+0x83f0): multiple definition of `Catch::leakDetector'
CMakeFiles/test_main.dir/main_at.cpp.o:(.bss+0x83f0): first defined here
CMakeFiles/test_main.dir/hello_at.cpp.o: In function `main':
/home/dheater/.conan/data/Catch2/2.11.0/catchorg/stable/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/catch2/catch.hpp:17161: multiple definition of `main'
CMakeFiles/test_main.dir/main_at.cpp.o:/home/dheater/.conan/data/Catch2/2.11.0/catchorg/stable/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/catch2/catch.hpp:17161: first defined here
collect2: error: ld returned 1 exit status
gmake[3]: *** [approval-tests/bin/approve] Error 1
gmake[2]: *** [approval-tests/CMakeFiles/approve.dir/all] Error 2
gmake[1]: *** [approval-tests/CMakeFiles/approve.dir/rule] Error 2
gmake: *** [approve] Error 2
```

Looks like this is the relevant snippet of code.
```
#if defined(APPROVALS_CATCH_EXISTING_MAIN)
    #define APPROVALS_CATCH
    #define CATCH_CONFIG_RUNNER
#elif defined(APPROVALS_CATCH)
    #define CATCH_CONFIG_MAIN
#endif

#ifdef APPROVALS_CATCH

#include <catch2/catch.hpp>
```

If I replace `APPROVALS_CATCH` with `APPROVALS_CATCH_EXISTING_MAIN` in `conanfile.py` that resolves the multiple definition of `main`, but the other functions still have multiple definitions.

> If I understand, you should define APPROVALS_CATCH only once, is that?
I think that is the root of the problem, yes.

Here is the documentation on how/when to use which version of `APPROVALS_CATCH...`: https://github.com/approvals/ApprovalTests.cpp/blob/master/doc/UsingCatch.md#top

Reply:
So let's remove the package_info from the recipe. What do you think?

Reply:
Agree